tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc

```diff
@@ -180,47 +180,68 @@ class CuckooHashTableOfTensorsGpu final : public LookupInterface {
   }
 
   void RehashIfNeeded(cudaStream_t stream) {
-    K* d_keys;
-    gpu::ValueArrayBase<V>* d_values;
+    RehashIfNeeded(stream, min_size_);
+  }
+
+  void RehashIfNeeded(cudaStream_t stream, size_t expecting) {
+    K* d_keys = nullptr;
+    gpu::ValueArrayBase<V>* d_values = nullptr;
     size_t* d_dump_counter;
-    size_t new_max_size = max_size_;
+    size_t capacity = table_->get_capacity();
+
+    size_t cur_size = table_->get_size(stream);
```

Comment:
from our profiling result, this kernel is expensive. I suggest that we can cache last size, and check if the table need to expand with expecting keys to be added. If not, we could add expect to last size, and just return. In this way, we could reduce the number of calls to get_size.

Reply:
Save and load are low-frequency operations for me. Is it frequently called in your application?